Makefile

```diff
@@ -1489,24 +1489,17 @@ cpplint: lint-cpp
 	$(warning Please use lint-cpp instead of cpplint)
 
 .PHONY: lint-py-build
-# python -m pip install flake8
+# python -m pip install ruff
 # Try with '--system' if it fails without; the system may have set '--user'
 lint-py-build:
-	$(info Pip installing flake8 linter on $(shell $(PYTHON) --version)...)
-	$(PYTHON) -m pip install --upgrade -t tools/pip/site-packages flake8 || \
-		$(PYTHON) -m pip install --upgrade --system -t tools/pip/site-packages flake8
+	$(info Pip installing ruff linter on $(shell $(PYTHON) --version)...)
+	$(PYTHON) -m pip install --upgrade --target tools/pip/site-packages ruff || \
+		$(PYTHON) -m pip install --upgrade --system -target tools/pip/site-packages ruff
 
 .PHONY: lint-py
-ifneq ("","$(wildcard tools/pip/site-packages/flake8)")
-# Lints the Python code with flake8.
-# Flag the build if there are Python syntax errors or undefined names
+# Lints the Python code with ruff.
 lint-py:
-	PYTHONPATH=tools/pip $(PYTHON) -m flake8 --count --show-source --statistics .
-else
-lint-py:
-	$(warning Python linting with flake8 is not available)
-	$(warning Run 'make lint-py-build')
-endif
+	ruff .
```

Comment:
We should keep (and adjust for this PR) the `ifneq` to indicate that `make lint-py-build` needs to be run if `ruff` is not installed.